src/components/tasks/card/index.tsx

```diff
@@ -313,11 +323,20 @@ const Card: FC<CardProps> = ({
         const data = {
             percentCompleted: percentCompleted,
         };
-        await updateTasks({
-            task: data,
-            id: id,
-        });
-        toast(SUCCESS, 'Progress Updated Successfully');
+        if (isUserAuthorized) {
+            await updateTask({
+                task: data,
+                id: id,
+            })
+                .unwrap()
+                .then(() => toast(SUCCESS, PROGRESS_SUCCESSFUL))
+                .catch(() => toast(ERROR, ERROR_MESSAGE));
+        } else {
+            await updateSelfTask({ task: data, id: id })
```

Comment:
same as above

Reply:
![image](https://github.com/Real-Dev-Squad/website-status/assets/82165483/347bd08d-8115-483c-bfae-36f230461c1e)

reference - https://redux-toolkit.js.org/rtk-query/usage/error-handling

This is used to handle RTK success and error status, please see the reference I have shared

Reply:
The example you mentioned doesn't use `await` as it is using `.then()` `.catch()` to handle the promise's result. In your case you're using both, `await` and `.then()` `.catch()`